feature/kl_divergence.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import NoReturn, Tuple
+
+import pandas as pd
+import numpy as np
+
+from scipy.special import rel_entr
+from feature.base import _BaseSupervisedSelector
+from feature.utils import Num, check_true
+
+
+class _KL_Divergence(_BaseSupervisedSelector):
+
+    def __init__(self, seed: int, num_features: Num, num_bins: Num = 100):
+        super().__init__(seed)
+
+        self.num_features = num_features  # this could be int or float
+        self.num_bins = num_bins
+
+    def fit(self, X: pd.DataFrame, y: pd.Series) -> NoReturn:
+
+        check_true(len(np.unique(y)) == 2, TypeError("Only binary labels are supported for KL Divergence"))
+        
+        kl_mat = np.zeros((self.num_features, 1))
+        X = X.values
+        label_categories = np.unique(y)
```

Comment:
move this before check_true to avoid np.unique() cost. labels column can be large